internal/attestation/initialize/initialize.go

```diff
@@ -0,0 +1,129 @@
+	// *os.File or an emulated device over a unix socket.
+	// Therefore, we can't simply use a type switch here,
+	// since the TPM may implement the same methods as the TDX device
+	if handle, ok := device.(tdx.Device); ok && tdx.IsTDXDevice(handle) {
+		return tdxMarkNodeAsBootstrapped(handle, clusterID)
+	}
+	return tpmMarkNodeAsBootstrapped(device, clusterID)
+}
+
+// IsNodeBootstrapped checks if a node is already bootstrapped by reading PCRs.
+func IsNodeBootstrapped(openDevice func() (io.ReadWriteCloser, error)) (bool, error) {
+	device, err := openDevice()
+	if err != nil {
+		return false, err
+	}
+	defer device.Close()
+
+	// The TDX device is of type *os.File, while the TPM device may be
+	// *os.File or an emulated device over a unix socket.
+	// Therefore, we can't simply use a type switch here,
+	// since the TPM may implement the same methods as the TDX device
+	if handle, ok := device.(tdx.Device); ok && tdx.IsTDXDevice(handle) {
+		return tdxIsNodeBootstrapped(handle)
+	}
+	return tpmIsNodeBootstrapped(device)
+}
+
+func tdxIsNodeBootstrapped(handle tdx.Device) (bool, error) {
+	tdMeasure, err := tdx.ReadMeasurements(handle)
+	if err != nil {
+		return false, err
+	}
+	return measurementInitialized(tdMeasure[measurements.RTMRIndexClusterID][:]), nil
+}
+
+func tpmIsNodeBootstrapped(tpm io.ReadWriteCloser) (bool, error) {
+	idxClusterID := int(measurements.PCRIndexClusterID)
+	pcrs, err := tpm2.ReadPCRs(tpm, tpm2.PCRSelection{
+		Hash: tpm2.AlgSHA256,
+		PCRs: []int{idxClusterID},
+	})
+	if err != nil {
+		return false, err
+	}
+	if len(pcrs[idxClusterID]) == 0 {
+		return false, errors.New("cluster ID PCR does not exist")
+	}
+	return measurementInitialized(pcrs[idxClusterID]), nil
+
+	/* Old code that will be reenabled in the future
+	idxOwner := int(PCRIndexOwnerID)
+	idxCluster := int(PCRIndexClusterID)
+	selection := tpm2.PCRSelection{
+		Hash: tpm2.AlgSHA256,
+		PCRs: []int{idxOwner, idxCluster},
+	}
+
+	pcrs, err := tpm2.ReadPCRs(tpm, selection)
+	if err != nil {
+		return false, err
+	}
+
+	if len(pcrs[idxOwner]) == 0 {
+		return false, errors.New("owner ID PCR does not exist")
+	}
+	if len(pcrs[idxCluster]) == 0 {
+		return false, errors.New("cluster ID PCR does not exist")
+	}
+
+	ownerInitialized := pcrInitialized(pcrs[idxOwner])
+	clusterInitialized := pcrInitialized(pcrs[idxCluster])
+
+	if ownerInitialized == clusterInitialized {
+		return ownerInitialized && clusterInitialized, nil
+	}
+	ownerState := "not initialized"
+	if ownerInitialized {
+		ownerState = "initialized"
+	}
+	clusterState := "not initialized"
+	if clusterInitialized {
+		clusterState = "initialized"
+	}
+	return false, fmt.Errorf("PCRs %v and %v are not consistent: PCR[%v]=%v (%v), PCR[%v]=%v (%v)", idxOwner, idxCluster, idxOwner, pcrs[idxOwner], ownerState, idxCluster, pcrs[idxCluster], clusterState)
+	*/
+}
+
+func tdxMarkNodeAsBootstrapped(handle tdx.Device, clusterID []byte) error {
+	return tdx.ExtendRTMR(handle, clusterID, 4)
```

Comment:
```suggestion
	return tdx.ExtendRTMR(handle, clusterID, 2)
```
`ExtendRTMR` has an index parameter that is unused so the 4 here does effectively nothing... But in any case that should be a 2 likely - there's no RTMR[4].